examples/moses-pln-synergy/scm/README.md

```diff
@@ -76,16 +76,204 @@ And search for the following
    )
 ...
 
-scheme@(guile-user)> (cog-bind pln-rule-eliminate-neutral-element-hack)
-scheme@(guile-user)> (cog-bind pln-rule-eliminate-dangling-junctor-hack)
-scheme@(guile-user)> (cog-bind pln-rule-equivalence-hack)
-$6 = (SetLink
+scheme@(guile-user)> (cog-bind implication-lambda-distribution-rule)
+...
    (ImplicationLink (stv 1 0.99999982)
-      (PredicateNode "take-treatment-1" (stv 0.1 0.80000001))
-      (PredicateNode "take-compound-A" (stv 0.2 0.80000001))
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (AndLink
+            (EvaluationLink
+               (PredicateNode "take")
+               (ListLink
+                  (VariableNode "$X")
+                  (ConceptNode "treatment-1")
+               )
+            )
+            (EvaluationLink (stv 1 0.99999982)
+               (PredicateNode "contain")
+               (ListLink
+                  (ConceptNode "treatment-1")
+                  (ConceptNode "compound-A")
+               )
+            )
+         )
+      )
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
```

Comment:
Well, this is a valid but somewhat funny way of saying things. The normal predicate-logic way of saying this is "for all x, if x takes treatment-1 and treament-1 contains compound-A then x takes compound-A"  There is a specific rewrite rule (I forget what its called, and I don't have my book on logic here with me) that allows you to say  "double-turnstile forall-x p(x) -> p(y) " (double-turnstile symbol because this is a semantic consequence, not a logical consequence).     

In ordinary predicate-logic p(y) always means lambda.y p(y) -- that is implicilty understood. So you converted a for-all rule into a specific rule.  Seeing the lambda there is .. correct, but unexpected and surprising, somehow.  It could use some articulation.